src/main/preprocessing/preprocessing.py

```diff
@@ -84,7 +99,11 @@ def preprocess_data(path):
     for folder in folders:
```

Comment:
It would be great to check if we have several files with the same 'unhashed' name. In that case, we should consider at_df as None